### PR TITLE
feat: use projen native TextFile features for generated files

### PR DESCRIPTION
**Pull request by hoegertn**
* **Please check if the PR fulfills these requirements**
- [X] The commit message describes your change
- [X] Tests for the changes have been added if possible (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Changes are mentioned in the changelog (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
cdk-serverless now uses projen's native feature of TextFile components instead of writing files to disk manually

* **What is the current behavior?** (You can also link to an open issue here)
cdk-serverless uses fs.writeFileSync to write generated files

* **What is the new behavior (if this is a feature change)?**
cdk-serverless uses pj.TextFile to generate files with the correct lifecycle

* **Does this PR introduce a breaking change?** (What changes might users need to make in their setup due to this PR?)

* **Other information**:

### CODE REVIEW

**Comment by hoegertn** [comment]
@DerDackel as mentioned I move the file generation to projen's native components.

No idea why I did not do this before.

File: src/projen/lazy-sampledir.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  /**
+   * The files to render into the directory.
+   */
+  readonly fileGenerator: () => {
+    [fileName: string]: () => string;
+  };
+}
+
+/**
+ * Renders the given files into the directory if the directory does not exist. Use this to create sample code files
```
Comment (hoegertn) [inline src/projen/lazy-sampledir.ts:18]:
```suggestion
 * Renders the given files into the directory if the files do not exist. Use this to create sample code files
```

File: src/projen/lazy-sampledir.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+/**
+ * Renders the given files into the directory if the files do not exist. Use this to create sample code files
+ */
+export class LazySampleDir extends Component {
+  private readonly dir: string;
+  private readonly options: LazySampleDirOptions;
+
+  /**
+   * Create sample files in the given directory if the given directory does not exist
```
Comment (hoegertn) [inline src/projen/lazy-sampledir.ts:25]:
```suggestion
   * Create sample files in the given directory
```

File: src/projen/lazy-sampledir.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+ * Renders the given files into the directory if the files do not exist. Use this to create sample code files
+ */
+export class LazySampleDir extends Component {
+  private readonly dir: string;
+  private readonly options: LazySampleDirOptions;
+
+  /**
+   * Create sample files in the given directory
+   * @param project Parent project to add files to.
+   * @param dir directory to add files to. If directory already exists, nothing is added.
```
Comment (hoegertn) [inline src/projen/lazy-sampledir.ts:27]:
```suggestion
   * @param dir directory to add files to.
```

**Review by DerDackel — APPROVED** [review]
LGTM

**Pull request merged**